src/tr1/game/lara/state.c

```diff
@@ -62,6 +65,32 @@ static int16_t M_FloorFront(ITEM *item, PHD_ANGLE ang, int32_t dist)
     return height;
 }
 
+static bool M_HasResponsiveState(const LARA_ANIMATION anim_idx)
+{
+    const OBJECT *const object = Object_GetObject(O_LARA);
+    if (!object->loaded) {
+        return false;
+    }
+
+    const ANIM *const anim = Object_GetAnim(object, anim_idx);
+    for (int32_t i = 0; i < anim->num_changes; i++) {
+        const ANIM_CHANGE *const change = Anim_GetChange(anim->change_idx + i);
+        if (change->goal_anim_state == LS_RESPONSIVE) {
+            return true;
+        }
+    }
+
+    return false;
+}
+
+void Lara_State_Initialise(void)
+{
+    m_EnableResponsiveJumping =
+        g_Config.gameplay.enable_tr2_jumping && M_HasResponsiveState(LA_RUN);
+    m_EnableResponsiveSwimming = g_Config.gameplay.enable_tr2_swim_cancel
+        && M_HasResponsiveState(LA_SWIM_FORWARD);
```

Comment:
Would it be possible to keep the `g_Config` property check for runtime? This way, we can ensure that the `/set` command can adjust the setting without needing to reload the level. While I'm not sure if this is how it currently works – for instance, using `/set braid on` does not take effect when the braid is turned off during loading the level, as the injections are skipped in that case – we can tackle that issue separately. My main point is to keep things forwards compatible, having this on our minds.

Reply:
That's a good point, thanks. Will adjust.

Reply:
Done. To confirm, switching the options at runtime does work in develop, so this retains that ability. Thanks again.

And yes, the braid issue could be tackled I think. We could always inject the required data, but create separate meshes for Lara's head and torso and store these somewhere. Then the option would determine which meshes to reference. I'll open a ticket.